### PR TITLE
Various small cleanups to PlatformBridgeCertValidator

### PR DESCRIPTION
**Pull request by RyanTheOptimist**
Remove unused next_iteration_callback_ member.
Remove unused config_ member.
Make some members const.
Use hostname instead of host_name.
Pass in the list of subject alt names instead of passing in the transport socket options.

Risk Level: Low
Testing: No behavior change
Docs Changes: N/A
Release Notes: N/A

Signed-off-by: Ryan Hamilton <rch@google.com>

### CODE REVIEW

**Review by jpsim — APPROVED** [review]

File: library/common/extensions/cert_validator/platform_bridge/platform_bridge_cert_validator.h [hunk 1]
```diff
@@ -56,22 +56,22 @@ class PlatformBridgeCertValidator : public CertValidator, Logger::Loggable<Logge
                     const Network::TransportSocketOptionsConstSharedPtr& transport_socket_options,
                     SSL_CTX& ssl_ctx,
                     const CertValidator::ExtraValidationContext& validation_context, bool is_server,
-                    absl::string_view host_name) override;
-  // As CA path will not be configured, make sure the return value won’t be SSL_VERIFY_NONE because
-  // of that, so that doVerifyCertChain() will be called from the TLS stack.
-  int initializeSslContexts(std::vector<SSL_CTX*> contexts,
-                            bool handshaker_provides_certificates) override;
+                    absl::string_view hostname) override;
+  // Return SSL_VERIFY_PEER so that doVerifyCertChain() will be called from the TLS stack.
```
Comment (danzh2010) [inline library/common/extensions/cert_validator/platform_bridge/platform_bridge_cert_validator.h:60]: nit: Returns

File: library/common/extensions/cert_validator/platform_bridge/platform_bridge_cert_validator.h [hunk 2]
```diff
@@ -83,26 +83,21 @@ class PlatformBridgeCertValidator : public CertValidator, Logger::Loggable<Logge
                                     uint8_t tls_alert, OptRef<Stats::Counter> error_counter);
 
   private:
-    Event::SchedulableCallbackPtr next_iteration_callback_;
```
Comment (danzh2010) [inline library/common/extensions/cert_validator/platform_bridge/platform_bridge_cert_validator.h:86]: nice catch!

File: library/common/extensions/cert_validator/platform_bridge/platform_bridge_cert_validator.h [hunk 3]
```diff
@@ -83,26 +83,21 @@ class PlatformBridgeCertValidator : public CertValidator, Logger::Loggable<Logge
...
+    const std::vector<envoy_data> certs_;
+    const std::string hostname_;
+    const std::vector<std::string> subject_alt_names_;
     Ssl::ValidateResultCallbackPtr result_callback_;
-    const Network::TransportSocketOptionsConstSharedPtr transport_socket_options_;
   };
 
-  // Calls into platform APIs in a stand-alone thread to verify the given certs.
-  // Once the validation is done, the result will be posted back to the current
-  // thread to trigger callback and update verify stats.
```
Comment (danzh2010) [inline library/common/extensions/cert_validator/platform_bridge/platform_bridge_cert_validator.h:96]: Can you move this comment to PengValidation::verifyCertsByPlatform()?

**Review by danzh2010 — COMMENTED** [review]
Thanks for the cleanup! Just a few nits left.

File: library/common/extensions/cert_validator/platform_bridge/platform_bridge_cert_validator.h (hunk 1 above)
Comment (RyanTheOptimist) [reply library/common/extensions/cert_validator/platform_bridge/platform_bridge_cert_validator.h]: Done!

File: library/common/extensions/cert_validator/platform_bridge/platform_bridge_cert_validator.h (hunk 2 above)
Comment (RyanTheOptimist) [reply library/common/extensions/cert_validator/platform_bridge/platform_bridge_cert_validator.h:86]: Thanks!

File: library/common/extensions/cert_validator/platform_bridge/platform_bridge_cert_validator.h (hunk 3 above)
Comment (RyanTheOptimist) [reply library/common/extensions/cert_validator/platform_bridge/platform_bridge_cert_validator.h:96]: Done!

**Review by RyanTheOptimist — COMMENTED** [review]
Thanks!

**Review by alyssawilk — APPROVED** [review]

**Pull request merged**